lib/linter/linter_component.coffee

```diff
@@ -0,0 +1,70 @@
+{Range, Point} = require 'atom'
+_ = require 'lodash'
+
+class LinterComponent
+  constructor: (@linter, @errorRepository) ->
+    LinterComponent.warn(@linter)
+    @setup() if @linter
+
+  @warn: (linter) ->
+    unless linter
+      msg = '[dart-tools] The linter package is required to provide details
+      about errors and warnings.'
+      atom.notifications.addWarning msg
+
+  check: => LinterComponent.warn(@linter)
+  setup: =>
+    # @linter.addLinter(linterConfig)
+
+    updateErrors = =>
+      messages = []
+
+      @errorRepository.forEachFileWithError (file, errors) =>
+        for error in errors
+          location = error.location
+          line      = location.startLine - 1
+          column    = location.startColumn - 1
+          range = new Range(
+            new Point(line, column),
+            new Point(line, column + location.length)
+          )
+
+          messages.push
+            type: _.capitalize(error.severity.toLowerCase())
+            text: error.message
+            filePath: file
+            range: range
+
+      @linter.deleteProjectMessages(linterConfig)
+      @linter.setProjectMessages linterConfig, messages
+      @linter.views?.render()
+
+    @errorRepository.onChange _.debounce(updateErrors, 250, maxWait: 1000)
+
+linterConfig =
+  grammarScopes: ['source.dart']
+  scope: 'project'
+  lintOnFly: false
+  # lint: ->
```

Comment:
I don't think `lint` gets exercised for project scope files (based on a cursory look at the codebase), and we don't add the linter  via `addLinter()`. Is there any benefit to doing so?